sphinx_gallery/gen_gallery.py

```diff
@@ -67,13 +67,9 @@ def clean_gallery_out(build_dir):
                 os.remove(os.path.join(build_image_dir, filename))
 
 
-def generate_gallery_rst(app):
-    """Generate the Main examples gallery reStructuredText
-
-    Start the sphinx-gallery configuration and recursively scan the examples
-    directories in order to populate the examples gallery
-    """
-    print('Generating gallery')
+def parse_config(app):
+    """Process the Sphinx Gallery configuration"""
+    # TODO: Test this behavior.
```

Comment:
@Titan-C sorry I don't have time to review this PR in detail so feel free to merge it. 

It does not have to happen in this PR but tests for the both old and new configuration are a must. It is very easy to break this kind of things without realizing and the impact on the user that encounters the regression is very bad IMO.